cmd/snap/cmd_find.go

```diff
@@ -113,39 +113,42 @@ func (x *cmdFind) Execute(args []string) error {
 	}
 
 	// magic! `snap find` returns the featured snaps
-	if x.Positional.Query == "" && x.Section == "" {
+	showFeatured := (x.Positional.Query == "" && x.Section == "")
+	if showFeatured {
 		x.Section = "featured"
 	}
 
-	return findSnaps(&client.FindOptions{
+	cli := Client()
+	opts := &client.FindOptions{
 		Private: x.Private,
 		Section: string(x.Section),
 		Query:   x.Positional.Query,
-	})
-}
-
-func findSnaps(opts *client.FindOptions) error {
-	cli := Client()
+	}
 	snaps, resInfo, err := cli.Find(opts)
 	if err != nil {
 		return err
 	}
-
 	if len(snaps) == 0 {
 		// TRANSLATORS: the %q is the (quoted) query the user entered
 		fmt.Fprintf(Stderr, i18n.G("The search %q returned 0 snaps\n"), opts.Query)
 		return nil
 	}
 
-	w := tabWriter()
-	defer w.Flush()
+	// show featured header *after* we checked for errors from the find
+	if showFeatured {
+		fmt.Fprintf(Stdout, i18n.G("No search term specified. Here are some interessting snaps:\n\n"))
```

Comment:
Typo on "interesting". Nice otherwise.

Since we need to touch the code anyway, can we change the message right above to "No matching snaps." or similar?  It seems a bit over verbose at the moment for something so straightforward and common.

Reply:
You should have said "Here are some pressious snaps", it could stay as an easter egg.